cmd/aws-vpc-cni/main.go

```diff
@@ -0,0 +1,410 @@
+	if podSGEnforcingMode != "strict" && podSGEnforcingMode != "standard" {
+		log.Errorf("%s must be set to either 'strict' or 'standard'", envPodSGEnforcingMode)
+		return false
+	}
+
+	prefixDelegationEn := getEnv(envEnPrefixDelegation, "false")
+	warmIPTarget := getEnv(envWarmIPTarget, "0")
+	warmPrefixTarget := getEnv(envWarmPrefixTarget, "0")
+	minimumIPTarget := getEnv(envMinIPTarget, "0")
+
+	if (prefixDelegationEn == "true") && (warmIPTarget <= "0" && warmPrefixTarget <= "0" && minimumIPTarget <= "0") {
+		log.Errorf("Setting WARM_PREFIX_TARGET = 0 is not supported while WARM_IP_TARGET/MINIMUM_IP_TARGET is not set. Please configure either one of the WARM_{PREFIX/IP}_TARGET or MINIMUM_IP_TARGET env variables")
+		return false
+	}
+	return true
+}
+
+func main() {
+	os.Exit(_main())
+}
+
+func _main() int {
+	log.Debug("Started aws-node container")
+	if !validateEnvVars() {
+		return 1
+	}
+
+	hostCNIBinPath := getEnv(envHostCniBinPath, defaultHostCNIBinPath)
+	configureRPFfilter := getEnv(envConfRPFfilter, "true")
+	if configureRPFfilter != "false" {
+		log.Infof("Copying CNI plugin binaries ... ")
+		pluginBins := []string{"loopback", "portmap", "bandwidth", "aws-cni-support.sh"}
+		err := cp.InstallBinaries(pluginBins, hostCNIBinPath)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to install binaries")
+			return 1
+		}
+	}
+
+	log.Infof("Install CNI binaries..")
+	pluginBins := []string{"aws-cni", "egress-v4-cni"}
+	err := cp.InstallBinaries(pluginBins, hostCNIBinPath)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to install CNI binaries")
+		return 1
+	}
+
+	log.Infof("Starting IPAM daemon... ")
+	agentLogPath := getEnv(envAgentLogPath, defaultAgentLogPath)
+
+	cmd := "./aws-k8s-agent"
+	ipamdDaemon := exec.Command(cmd, "|", "tee", "-i", agentLogPath, "2>&1")
+	err = ipamdDaemon.Start()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to execute command: %s", cmd)
+		return 1
+	}
+
+	log.Infof("Checking for IPAM connectivity... ")
+	if !waitForIPAM() {
+		log.Errorf("Timed out waiting for IPAM daemon to start")
+
+		byteValue, err := ioutil.ReadFile(agentLogPath)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to read %s", agentLogPath)
+		}
+		log.Infof("%s", string(byteValue))
+		return 1
+	}
+
+	// Wait for init container to complete
+	//if err = waitForInit(); err != nil {
+	//	log.WithError(err).Errorf("Init container failed to complete")
+	//	return 1
+	//}
+
+	log.Infof("Copying config file... ")
+	err = generateJSON(defaultAWSconflistFile, tmpAWSconflistFile)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to update 10-awsconflist")
+		return 1
+	}
+
+	err = cp.CopyFile(tmpAWSconflistFile, defaultHostCNIConfDirPath+awsConflistFile)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to update 10-awsconflist")
+		return 1
+	}
+
+	log.Infof("Successfully copied CNI plugin binary and config file.")
+
+	awsConfFile := defaultHostCNIConfDirPath + "/aws.conf"
```

Comment:
I wonder if we really need this anymore. v1.6.0+ anyways migrated to `10-aws.conflist` and it has been 2+ years already.

Reply:
The only thing we are doing here is deleting this file if it exists. But since no one consumes it and we do not allow upgrades from version < `v1.6.0` to `v1.12.1`, it does seem like this can be deleted. I'll test it out now

Reply:
removed in latest revision